vkaci pod constantly crashes

The vkaci pod crashes, backs off, tries again and keeps crashing.  

```
cisco@nkt-master-1:~$ kubectl get pods
NAME                     READY   STATUS             RESTARTS       AGE
goldpinger-brmtd         1/1     Running            1              39m
goldpinger-l4l4j         1/1     Running            1              39m
goldpinger-z2hpr         1/1     Running            1              39m
my-neo4j-0               1/1     Running            1              39m
vkaci-547fc784f9-wbswf   0/1     CrashLoopBackOff   15 (62s ago)   39m
```
logs show this condition...maybe something to do with LLDP

```
2022-10-07 11:50:38,916 graph INFO [ThreadPoolExecutor-0_1] Interface topology/pod-1/paths-2102/pathep-[eth1/31] is Bundled under topology/pod-1/protpaths-2101-2102/pathep-[VPC_KUBE02]
2022-10-07 11:50:38,986 graph INFO [ThreadPoolExecutor-0_0] Found the following Host as Neighbour
2022-10-07 11:50:38,987 graph INFO [ThreadPoolExecutor-0_0] Found leaf-2101 as Neighbour to :
2022-10-07 11:50:39,024 graph INFO [ThreadPoolExecutor-0_1] Found the following Host as Neighbour
2022-10-07 11:50:39,024 graph INFO [ThreadPoolExecutor-0_1] Found leaf-2101 as Neighbour to :
2022-10-07 11:50:39,026 graph INFO [ThreadPoolExecutor-0_2] Found the following Host as Neighbour
2022-10-07 11:50:39,026 graph INFO [ThreadPoolExecutor-0_2] Found leaf-2101 as Neighbour to :
[2022-10-07 11:50:39 +0000] [7] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/gunicorn/arbiter.py", line 589, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/ggevent.py", line 146, in init_process
    super().init_process()
  File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.10/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.10/site-packages/gunicorn/util.py", line 359, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/app/visibility_ui.py", line 15, in <module>
    graph.update_database()
  File "/app/graph.py", line 635, in update_database
    topology = self.topology.update()
  File "/app/graph.py", line 521, in update
    result = future.result()
  File "/usr/local/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/usr/local/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/app/graph.py", line 377, in update_node
    self.add_neighbour(node, lldp_neighbour)
  File "/app/graph.py", line 221, in add_neighbour
    if 'VMware' in neighbour_description:
TypeError: argument of type 'NoneType' is not iterable
```
Checking physical interface on the ESXi host where all this runs, I do see that LLDP is working




Link Layer Discovery Protocol from vCenter View
--
```
Chassis ID | 00:3a:9c:7b:f0:9f
Port ID | Eth1/31
Time to live | 117
TimeOut | 30
Samples | 39
Management Address | 10.50.129.212
Port Description | topology/pod-1/protpaths-2101-2102/pathep-[VPC_KUBE02]
System Description | topology/pod-1/node-2101
System Name | leaf-2101
```

Happy to provide lab access for internal people.

--- a/app/graph.py
+++ b/app/graph.py
@@ -218,10 +218,10 @@ class VkaciBuilTopology(object):
             
             #Vmare esxi puts the port name vmnic in the chassisIdV while linux put the port name in the portDesc
             lldp_port_id_class = 'portDesc'
-            if 'VMware' in neighbour_description:
+            if neighbour_description and 'VMware' in neighbour_description:
                 lldp_port_id_class = 'chassisIdV'
             # UCS Uses the portIdV
-            if 'Cisco' in neighbour_description:
+            if neighbour_description and 'Cisco' in neighbour_description:
                 lldp_port_id_class = 'portIdV'
             
             neighbour_adj_port = getattr(neighbour_adj,lldp_port_id_class, None)

--- a/unit_test.py
+++ b/unit_test.py
@@ -58,7 +58,7 @@ class MockMo(object):
         self.fvRsCEpToPathEp[0].tDn = pathtDn
 
 
-def create_lldp_neighbour(on: bool = True):
+def create_lldp_neighbour(on: bool = True, desc: bool = True):
     n = Expando()
     n.operTxSt = n.operRxSt = "down"
     if (on):
@@ -66,7 +66,8 @@ def create_lldp_neighbour(on: bool = True):
         n.lldpAdjEp = [Expando()]
         n.lldpAdjEp[0].sysName = "esxi4.cam.ciscolabs.com"
         n.lldpAdjEp[0].chassisIdV = "vmxnic1"
-        n.lldpAdjEp[0].sysDesc = "VMware version 123"
+        if desc:
+            n.lldpAdjEp[0].sysDesc = "VMware version 123"
         n.sysDesc = n.dn = "topology/pod-1/node-204"
         n.id = "eth1/1"
     return n
@@ -260,6 +261,37 @@ class TestVkaciGraph(unittest.TestCase):
             VkaciEnvVariables(self.vars), mock)
         # Act
         result = build.update()
+        # Assert
+        self.assertDictEqual(result, expected)
+        self.assertEqual(build.aci_vrf, "uni/tn-Ciscolive/ctx-vrf-01")
+
+
+    def test_valid_topology_no_desc_neighbour(self):
+        """Test that a neighbour with no description will not crash"""
+        # Arrange
+        expected = {'nodes': {'1234abc': {'bgp_peers': {'leaf-204': {'prefix_count': 2}},
+                                          'labels': {'app': 'redis'},
+                                          'mac': 'MOCKMO1C',
+                                          'neighbours': {'esxi4.cam.ciscolabs.com': {'Description': '',
+                                                                        'switches': {'leaf-204': set()}}},
+                                          'node_ip': '192.168.1.2',
+                                          'pods': {'dateformat': {'ip': '192.158.1.3',
+                                                                  'labels': {'guest': 'frontend'},
+                                                                  'ns': 'dockerimage'}}}},
+                    'services': {'appx': [{'cluster_ip': '192.168.25.5',
+                                           'external_i_ps': ['192.168.5.1'],
+                                           'labels': {'app': 'guestbook'},
+                                           'name': 'example service',
+                                           'prefix': '192.168.5.1/32'}]}}
+
+        mock = ApicMethodsMock()
+        mock.lldps = [create_lldp_neighbour(desc=False)]
+        mock.cdpns = []
+        build = VkaciBuilTopology(
+            VkaciEnvVariables(self.vars), mock)
+        # Act
+        result = build.update()
+        print(result)
         # Assert
         self.assertDictEqual(result, expected)
         self.assertEqual(build.aci_vrf, "uni/tn-Ciscolive/ctx-vrf-01")

